CHANGES.rst

```diff
@@ -390,6 +390,10 @@ Other Changes and Additions
     ``angles.angle_axis`` are also deprecated, in favour of the new routines
     with the same name in ``coordinates.matrix_utilities``. [#5104]
 
+  - A new ``BaseCoordinateFrame.cache`` dictionary has been created to expose
+    the internal cache. This is useful when modifying representation data
+    inplace without using ``realize_frame`` [#5575]
```

Comment:
Needs a period at the end of the sentence.  Then add one more sentence "Added new documentation describing details of such inplace modification of an existing coordinate object."  Though we don't typically mention doc updates in the changelog, in this case it seems worthwhile.